src/python/grpcio_tests/tests_aio/unit/client_unary_stream_interceptor_test.py

```diff
@@ -0,0 +1,404 @@
+class _UnaryStreamInterceptorEmpty(aio.UnaryStreamClientInterceptor):
+
+    async def intercept_unary_stream(self, continuation, client_call_details,
+                                     request):
+        return await continuation(client_call_details, request)
+
+
+class _UnaryStreamInterceptorWith_ResponseIterator(
+        aio.UnaryStreamClientInterceptor):
+
+    def __init__(self):
+        self.response_iterator = None
+
+    async def intercept_unary_stream(self, continuation, client_call_details,
+                                     request):
+        call = await continuation(client_call_details, request)
+        self.response_iterator = _ResponseIterator(call)
+        return self.response_iterator
+
+
+class TestUnaryStreamClientInterceptor(AioTestBase):
+
+    async def setUp(self):
+        self._server_target, self._server = await start_test_server()
+
+    async def tearDown(self):
+        await self._server.stop(None)
+
+    async def test_intercepts(self):
+        for interceptor_class in (_UnaryStreamInterceptorEmpty,
+                                  _UnaryStreamInterceptorWith_ResponseIterator):
+
+            with self.subTest(name=interceptor_class):
+                interceptor = interceptor_class()
+
+                request = messages_pb2.StreamingOutputCallRequest()
+                for _ in range(_NUM_STREAM_RESPONSES):
+                    request.response_parameters.append(
+                        messages_pb2.ResponseParameters(
+                            size=_RESPONSE_PAYLOAD_SIZE))
+
+                channel = aio.insecure_channel(self._server_target,
+                                               interceptors=[interceptor])
+                stub = test_pb2_grpc.TestServiceStub(channel)
+                call = stub.StreamingOutputCall(request)
+
+                await call.wait_for_connection()
+
+                response_cnt = 0
+                async for response in call:
+                    response_cnt += 1
+                    self.assertIs(type(response),
+                                  messages_pb2.StreamingOutputCallResponse)
+                    self.assertEqual(_RESPONSE_PAYLOAD_SIZE,
+                                     len(response.payload.body))
+
+                self.assertTrue(response_cnt, _NUM_STREAM_RESPONSES)
+                self.assertEqual(await call.code(), grpc.StatusCode.OK)
+                self.assertEqual(await call.initial_metadata(), ())
+                self.assertEqual(await call.trailing_metadata(), ())
+                self.assertEqual(await call.details(), '')
+                self.assertEqual(await call.debug_error_string(), '')
+                self.assertEqual(call.cancel(), False)
+                self.assertEqual(call.cancelled(), False)
+                self.assertEqual(call.done(), True)
+
+                if interceptor_class == _UnaryStreamInterceptorWith_ResponseIterator:
+                    self.assertTrue(interceptor.response_iterator.response_cnt,
```

Comment:
`assertEqual`?

Also, it will be surprising if this fails due to a lack of a `response_cnt` member on `response_iterator`. Can we add an assertion about the type of `interceptor.response_iterator`?

Reply:
facepalm